packages/docusaurus-plugin-client-redirects/src/utils.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+
+export function addTrailingSlash(str: string) {
+  return str.endsWith('/') ? '' : '/';
+}
+
+export function removeTrailingSlash(str: string) {
+  return str.endsWith('/') ? str.substr(0, str.length - 2) : str;
```

Comment:
Could we do this?
`str.substr(0, str.length - 2)` -> `str.slice(0, -1)`

It should be faster.

Reply:
ah yes, always forget we can write this :)